pzsd_bot/__main__.py

```diff
@@ -1,32 +1,47 @@
+import asyncio
 import logging
 import os
-from pathlib import Path
+import re
+from datetime import datetime
 
 import discord
-import yaml
+from discord import Intents
 from dotenv import load_dotenv
+from sqlalchemy import insert, select
+from sqlalchemy.exc import NoResultFound
+from sqlalchemy.ext.asyncio import create_async_engine
 
-config_path = Path(__file__).parent / 'config.yml'
-if config_path.exists():
-    with config_path.open() as f:
-        config = yaml.safe_load(f)
-else:
-    config = {}
-
-LOG_LEVEL = config.get('LOG_LEVEL', 'INFO')
+from pzsd_bot.model import ledger, pzsd_user
 
 load_dotenv()
 
-BOT_TOKEN = os.environ['BOT_TOKEN']
-GUILD_ID = os.environ['GUILD_ID']
+LOG_LEVEL = os.getenv("LOG_LEVEL", "INFO")
+BOT_TOKEN = os.environ["BOT_TOKEN"]
+GUILD_ID = os.environ["GUILD_ID"]
+
+connection_str = "postgresql+asyncpg://{}:{}@{}:{}/{}".format(
+    os.getenv("PGUSER", "postgres"),
+    os.getenv("PGPASSWORD", "password"),
+    os.getenv("PGHOST", "localhost"),
+    os.getenv("PGPORT", "5432"),
+    os.getenv("PGDATABASE", "pzsd"),
+)
+engine = create_async_engine(connection_str)
 
-logger = logging.getLogger('pzsd')
+logger = logging.getLogger("pzsd")
 logger.setLevel(LOG_LEVEL)
-handler = logging.FileHandler(filename='pzsd_bot.log', encoding='utf-8', mode='w')
-handler.setFormatter(logging.Formatter('%(asctime)s:%(levelname)s:%(name)s: %(message)s'))
+handler = logging.FileHandler(filename="pzsd_bot.log", encoding="utf-8", mode="w")
+handler.setFormatter(
+    logging.Formatter("%(asctime)s:%(levelname)s:%(name)s: %(message)s")
+)
 logger.addHandler(handler)
 
-bot = discord.Bot()
+bot = discord.Bot(intents=Intents.all())
+
+point_pattern = re.compile(
+    r"(?:^| )([+-]?(?:\d+|\d{1,3}(?:,\d{3})*)) +points? (?:to|for) (?:(\w+)|<@(\d+)>)",
```

Comment:
Use named match groups for readability and to make retrieving match groups more explicit
```python
r"(?:^| )([+-]?(?:\d+|\d{1,3}(?:,\d{3})*)) +points? (?:to|for) (?:(\w+)|<@(\d+)>)",
```

Reply:
I do like named match groups. I find the reason I never use them is because it makes the regex super long. But I think it's worth it for the readability, and this regex is already long anyway. Perhaps I'll eventually create a module just for constants and put verbose stuff like this in there.